lib/itamae/plugin/recipe/mackerel-agent/default.rb

```diff
@@ -7,24 +7,55 @@
 node['mackerel-agent']['package-action'] = node['mackerel-agent'].fetch('package-action', :install)
 node['mackerel-agent']['plugins'] = node['mackerel-agent'].fetch('plugins', [])
 
+def platform_version_satisfy?(op)
+  Gem::Requirement.create(op).satisfied_by?(Gem::Version.create(node[:platform_version]))
+end
+
 case node[:platform]
 when "debian", "ubuntu"
-  remote_file "/etc/apt/sources.list.d/mackerel.list"
-  execute "import mackerel GPG key" do
-    command "curl -fsS https://mackerel.io/file/cert/GPG-KEY-mackerel | apt-key add -"
+  if node[:platform] == "debian" and platform_version_satisfy?('>= 8') or node[:platform] == "ubuntu" and platform_version_satisfy?('>= 16.04')
+    execute "import mackerel GPG key v2" do
+      command "curl -fsS https://mackerel.io/file/cert/GPG-KEY-mackerel-v2 | apt-key add -"
```

Comment:
[nit]
It's happy for us (mackerel.io provider) if this `curl` has `-L` option so that we can change this URL with redirecting in future.